packages/modules/workflow-engine-inmemory/src/utils/workflow-orchestrator-storage.ts

```diff
@@ -383,16 +383,31 @@ export class InMemoryDistributedTransactionStorage
     // In order to ensure that the schedule configuration is always up to date, we first cancel an existing job, if there was one
     // any only then we add the new one.
     await this.remove(jobId)
-    const expression = parseExpression(schedulerOptions.cron)
-    const nextExecution = expression.next().getTime() - Date.now()
+    const expression =
+      "cron" in schedulerOptions ? parseExpression(schedulerOptions.cron) : null
+    let nextExecution = 0
+    if (expression) {
+      nextExecution = expression.next().getTime() - Date.now()
+    } else if ("interval" in schedulerOptions) {
+      nextExecution = schedulerOptions.interval
+    } else {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Cron schedule cron or interval definition is required for scheduled jobs."
+      )
+    }
 
     const timer = setTimeout(async () => {
       this.jobHandler(jobId)
     }, nextExecution)
 
     this.scheduled.set(jobId, {
       timer,
-      expression,
+      expression:
+        expression ??
```

Comment:
Maybe we can move the `interval:..` part to the expression definition above? So above we'd have

```
let expression
if ("cron" in schedulerOptions) {
 expression = ...
} else if ("interval" in schedulerOptions {
expression = ...
} else throw

const nextExecution = getNextExecution(expression)
```

Reply:
I ve done all the changes [here](https://github.com/medusajs/medusa/pull/11800/commits/8ed96fea0915f69eae3a386923cd625ac04fbfb8) let me know wyt 👍